### PR TITLE
refactor(client) ♻️ put timer in game page 

### LINKED ISSUE #141: Refactor timers to reduce code duplication
Currently the timer is implemented on every page. this leads to a lot of duplicated code.

tasks:
- [ ] refactor the the timer to be only in one place by moving code to the game page

acceptance criteria:
- the code for displaying a timer for the expiration of a game state is not duplicated between components

### PR DESCRIPTION
**Pull request by gchamb**
- puts the timer in one spot on the game page
- removes timers from other components
- changed chatroom style since removing the p tag messed it up

### CODE REVIEW

**Review by mrjoshua520 — DISMISSED** [review]

**Review by nstringham — DISMISSED** [review]
please pull from dev

File: functions/src/firestore-types/lobby.ts [hunk 1]
```diff
@@ -51,6 +51,19 @@ export const GAME_STATE_DURATIONS: { [state in GameState]: number } = {
...
+ */
+export const DISPLAY_TIMERS: { [state in GameState]: boolean | null } = {
+  WAIT: null,
+  ROLE: false,
+  PROMPT: true,
+  CHAT: true,
+  VOTE: true,
+  RESULT: false,
+  END: false,
+};
```
Comment (nstringham) [inline functions/src/firestore-types/lobby.ts:65]: This file is for things in the database and DISPLAY_TIMERS seams to only be relivant to the client

File: functions/src/firestore-types/lobby.ts [hunk 2]
```diff
@@ -51,6 +51,19 @@ export const GAME_STATE_DURATIONS: { [state in GameState]: number } = {
   END: 10,
 };
 
+/**
+ * whether to show timer or not
+ */
+export const DISPLAY_TIMERS: { [state in GameState]: boolean | null } = {
+  WAIT: null,
```
Comment (nstringham) [inline functions/src/firestore-types/lobby.ts]: what is the diffrence between null and false

File: functions/src/firestore-types/lobby.ts (hunk 1 above)
Comment (nstringham) [inline functions/src/firestore-types/lobby.ts]: This file is for things in the database and DISPLAY_TIMERS seams to only be relivant to the client

File: client/src/routes/game/+page.svelte [hunk 3]
```diff
@@ -59,7 +69,17 @@
     // We want them to subscribe to the lobby on mount
     unsubscribeLobby = onSnapshot(lobbyDocRef, (doc) => {
       // will change lobby to the new doc data
-      lobby = doc.data();
+      const newLobby = doc.data();
+      if (lobby?.state != newLobby?.state && newLobby?.state != null) {
```
Comment (nstringham) [inline client/src/routes/game/+page.svelte]: `newLobby?.state != null`

is state ever null?

File: client/src/routes/game/+page.svelte [hunk 4]
```diff
@@ -59,7 +69,17 @@
...
+        countdown = GAME_STATE_DURATIONS[newLobby?.state];
```
Comment (nstringham) [inline client/src/routes/game/+page.svelte]: why this `?`

File: client/src/routes/game/+page.svelte [hunk 5]
```diff
@@ -24,6 +31,9 @@
   let privatePlayer: PrivatePlayer | undefined = undefined;
   let unsubscribePrivatePlayer: Unsubscribe | undefined = undefined;
 
+  let countdown = GAME_STATE_DURATIONS.WAIT;
+  let timer: ReturnType<typeof setInterval>;
```
Comment (nstringham) [inline client/src/routes/game/+page.svelte:31]: timer is never cleared

File: client/src/routes/game/+page.svelte [hunk 6]
```diff
@@ -59,7 +69,17 @@
...
-      lobby = doc.data();
+      const newLobby = doc.data();
+      if (lobby?.state != newLobby?.state && newLobby?.state != null) {
+        countdown = GAME_STATE_DURATIONS[newLobby?.state];
+        timer = setInterval(() => {
+          if (lobby?.expiration != undefined) {
+            const diff = Math.floor((lobby.expiration.toMillis() - Date.now()) / 1000);
+            countdown = diff;
+          }
+        }, 500);
```
Comment (nstringham) [inline client/src/routes/game/+page.svelte:78]: probobly shouldn't be in onSnapshot

File: client/src/routes/game/+page.svelte [hunk 7]
```diff
@@ -122,6 +158,10 @@
 </main>
 
 <style>
+  .countdown {
+    margin-top: 0;
+    text-align: center;
+  }
   main {
```
Comment (nstringham) [inline client/src/routes/game/+page.svelte]: can we have a blank line please

File: client/src/components/ChatRoom.svelte [hunk 8]
```diff
@@ -148,15 +117,11 @@
...
-  }
-
-  .countdown {
-    margin: 0;
+    display: flex;
+    justify-content: center;
+    height: 80%;
+    width: 75%;
+    margin: auto;
   }
```
Comment (nstringham) [inline client/src/components/ChatRoom.svelte]: this looks very bad on iPhone SE

**Review by nstringham — DISMISSED** [review]

File: client/src/components/Vote.svelte [hunk 9]
```diff
@@ -40,26 +29,12 @@
...
 </script>
 
 <div class="lobby-chat-level">
   <LobbyChat {lobby} {lobbyCode} />
 </div>
 <div class="voting">
-  <p class="countdown mdc-typography--headline2 {countdown < 10 ? 'error' : ''}">
-    {formatTimer(Math.max(countdown, 0))}
-  </p>
-  <p class="mdc-typography--headline4">Vote out the catfish</p>
```
Comment (nstringham) [inline client/src/components/Vote.svelte:58]: you sure you wanna do that?

**Review by nstringham — DISMISSED** [review]
the CSS is a little jank but we will fix it later

**Review by nstringham — APPROVED** [review]

**Pull request merged**